### PR TITLE
RPS-435 - Future RPS journey + ability to fetch data from CV refreshing a token

### PR DESCRIPTION
**Pull request by dcapilla**

### CODE REVIEW

File: src/config/index.ts [hunk 1]
```diff
@@ -58,7 +58,46 @@ export const config = convict({
     default: isProduction,
     env: 'ENFORCE_CSRF'
   },
-
+  entra: {
+    tokenEndpoint: {
+      doc: 'Microsoft entra token endpoint',
+      format: String,
+      default: '',
+      env: 'ENTRA_INTERNAL_TOKEN_URL'
```
Comment (Andrew-Folga) [inline src/config/index.ts:66]: Wouldn't we be required to use AWS Cognito for DEFRA ID integraiton?

Comment (dcapilla) [reply src/config/index.ts:66]: @Andrew-Folga - in this specific scenario we're identifying our service against Consolidated View service, which is on Azure. For those services that are on AWS I'd assume we need to do something similar but with AWS credentials.

Regarding Defra ID integration yes, we need to do that. But that's for authenticating users, this is to authenticate our application calling other services.

File: src/server/utils/token-manager.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+ * @property {number|null} tokenExpiry - Expiry time of the token in milliseconds
+ */
+
+/** @type {TokenState} */
+let tokenState = {
+  currentToken: null,
+  tokenExpiry: null
+}
+
+export function clearTokenState() {
```
Comment (Andrew-Folga) [inline src/server/utils/token-manager.js:21]: Do we not want to explicitly test this function?

Comment (Andrew-Folga) [reply src/server/utils/token-manager.js:21]: Oh so this was added in source code just for the tests I see.

File: src/server/utils/token-manager.test.js [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+      const bodyParams = new URLSearchParams(calledOptions.body)
+      expect(Object.fromEntries(bodyParams)).toEqual({
+        client_id: 'mock-client-id',
+        scope: 'mock-client-id/.default',
+        client_secret: 'mock-client-secret',
+        grant_type: 'client_credentials'
+      })
+    })
+
+    test('throws error when token refresh fails', async () => {
```
Comment (Andrew-Folga) [inline src/server/utils/token-manager.test.js:97]: Would we need a test for the case when API returns an empty response or malformed JSON?

Comment (dcapilla) [reply src/server/utils/token-manager.test.js:97]: added, thanks

File: src/server/common/helpers/consolidated-view/consolidated-view.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    if (!response.ok) {
+      /**
+       * @type {Error & {code?: number}}
+       */
+      const error = new Error(response.statusText)
+      error.code = response.status
+      throw error
+    }
+  } catch (error) {
+    logger.error(error, `Failed to refresh token`)
```
Comment (Andrew-Folga) [inline src/server/common/helpers/consolidated-view/consolidated-view.js:68]: Would this error message be correct or should it be more about the business details etc?

Comment (dcapilla) [reply src/server/common/helpers/consolidated-view/consolidated-view.js]: updated, thanks

**Review by Andrew-Folga — APPROVED** [review]
LGTM
